src/libsyntax/parse/parser.rs

```diff
@@ -6172,6 +6174,45 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_expansion(&mut self, kind: ExpansionKind) -> PResult<'a, Expansion> {
```

Comment:
I wonder if this might be better in ext somewhere rather than in the Parser. IIRC the parser is fairly ignorant of macro stuff at the moment but is kind of huge and hard to understand, I think it might serve us well to minimise adding macro stuff. What do you think?